pkg/scalers/azure_data_explorer_scaler.go

```diff
@@ -0,0 +1,198 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	"k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/scalers/azure"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type azureDataExplorerScaler struct {
+	metadata  *azure.DataExplorerMetadata
+	client    *kusto.Client
+	name      string
+	namespace string
+}
+
+const adxName = "azure-data-explorer"
+
+var dataExplorerLogger = logf.Log.WithName("azure_data_explorer_scaler")
+
+func NewAzureDataExplorerScaler(config *ScalerConfig) (Scaler, error) {
+	metadata, err := parseAzureDataExplorerMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse azure data explorer metadata: %s", err)
+	}
+
+	client, err := azure.CreateAzureDataExplorerClient(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create azure data explorer client: %s", err)
+	}
+
+	return &azureDataExplorerScaler{
+		metadata:  metadata,
+		client:    client,
+		name:      config.Name,
+		namespace: config.Namespace,
+	}, nil
+}
+
+func parseAzureDataExplorerMetadata(config *ScalerConfig) (*azure.DataExplorerMetadata, error) {
+	metadata, err := parseAzureDataExplorerAuthParams(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get database name.
+	databaseName, err := getParameterFromConfig(config, "databaseName", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.DatabaseName = databaseName
+
+	// Get endpoint.
+	endpoint, err := getParameterFromConfig(config, "endpoint", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.Endpoint = endpoint
+
+	// Get query.
+	query, err := getParameterFromConfig(config, "query", false)
+	if err != nil {
+		return nil, err
+	}
+	metadata.Query = query
+
+	// Get threshold.
+	if val, ok := config.TriggerMetadata["threshold"]; ok {
+		threshold, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing metadata. Details: can't parse threshold. Inner Error: %v", err)
+		}
+		metadata.Threshold = threshold
+	}
+
+	// Resolve metricName.
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		metadata.MetricName = kedautil.NormalizeString(fmt.Sprintf("%s-%s", adxName, val))
```

Comment:
We prefer to not expose `metricName` to the scaler config, some older scalers is doing that, but any new one shouldn't do that. It is internal implementation detail and users shouldn't be concerned. Please remove this.

Reply:
done